src/ResourceManager/Profile/Commands.Profile/Context/GetAzureRMContext.cs

```diff
@@ -26,6 +27,26 @@ namespace Microsoft.Azure.Commands.Profile
     [OutputType(typeof(PSAzureContext))]
     public class GetAzureRMContextCommand : AzureRMCmdlet
     {
+        /// <summary>
+        /// Gets the current default context.
+        /// </summary>
+        protected override AzureContext DefaultContext
+        {
+            get
+            {
+                if (DefaultProfile == null || DefaultProfile.Context == null)
+                {
+                    WriteError(new ErrorRecord(
+                        new PSInvalidOperationException("Run Login-AzureRmAccount to login."),
+                        string.Empty,
+                        ErrorCategory.AuthenticationError,
+                        null));
+                }
+
+                return DefaultProfile.Context;
```

Comment:
@cormacpayne also is there a reason, we are not replacing the throw Exception in the base class with WriteError.
Do you know under any circumstances if we simply have it in the base class won't solve this issue?